ecpy_pulses/workspace/dialogs.enaml

```diff
@@ -0,0 +1,357 @@
+            DictEditor(_ValEditor): de:
+                de.mapping := workspace.state.sequence.external_vars
+                de.operations = ['move']
+                
+#            DictEditor(_ValEditor): ed:
+#                ed.model = workspace.state.sequence
+#                ed.iterable_name = 'external_vars'
+#                ed.title = 'Variables'
+#                ed.operations = ['move']
+
+    Label: dur_lab:
+        text = 'Compilation duration (µs)'
+
+    Field: dur_val:
+        read_only = True
+        text << unicode(elapsed_time)
+
+    PushButton: comp:
+        text = 'Compile'
+        clicked ::
+            seq = workspace.state.sequence
+            start = time()
+            res, report = seq.compile_sequence(use_context=use_context)
+            stop = time()
+            main.elapsed_time = (stop - start)*1e6
+
+            if not res:
+                dur_val.background = 'red'
+                miss, err = report
+                mess = 'The values of those variables were never computed : {}'
+                if miss:
+                    err['Undefined variables'] = mess.format(miss)
+                main.errors = err
+                outcome(False)
+            else:
+                dur_val.background = 'green'
+                outcome(True)
+
+
+TYPE_EXPLANATION = \
+'''Standard mode :
+- the sequence is saved as is and can be used in the measurement program
+directly. It should have a valid context (matching the instrument to be used).
+The value of the external vars are specified in the measurement program.
+Template mode :
+- the sequence is a model that can be reused in other sequences, to make
+sequence development easier and avoid duplicating standard sequences. The only
+accessible part of a template sequence are the template vars and the
+TemplateContext which specifies how to map the real channels to the one defined
+in the template.'''
+
+enamldef TypeSelectionDialog(Dialog): dial:
+    """ Dialog allowing to choose how to save the edited sequence.
+
+    """
+    attr type
+    title = 'How to save sequence ?'
+    initial_size = (700, 250)
+
+    Container:
+        padding = 2
+        constraints = [vbox(doc, hbox(canc, spacer, temp, std))]
+
+        MultilineField: doc:
+            read_only = True
+            text = TYPE_EXPLANATION
+        PushButton: canc:
+            text = 'Cancel'
+            clicked :: dial.reject()
+        PushButton: temp:
+            text = 'Template'
+            clicked ::
+                dial.type = 'template'
+                dial.accept()
+        PushButton: std:
+            text = 'Standard'
+            clicked ::
+                dial.type = 'file'
+                dial.accept()
+
+
+enamldef TemplateLoadDialog(Dialog): dial:
+    """ Dialog used to load a template sequence for edition.
+
+    """
+    alias manager : selector.manager
+    attr doc = ''
+    attr prefs
+    attr t_infos
+
+    title = 'Load template.'
+
```

Comment:
blankline and new member/alias should be before teh ones whose default value is simply set.